lib/ae_mdw/sync/server.ex

```diff
@@ -345,14 +387,20 @@ defmodule AeMdw.Sync.Server do
   end
 
   defp last_microblock(blocks_mutations) do
-    {block_index, mblock, _mutations} =
-      blocks_mutations
-      |> Enum.filter(fn {{_height, mbi}, _block, _mutations} -> mbi != -1 end)
-      |> Enum.max_by(fn {block_index, _block, _mutations} -> block_index end)
+    blocks_mutations
+    |> Enum.filter(fn {{_height, mbi}, _block, _mutations} -> mbi != -1 end)
+    |> case do
+      [] ->
+        {:error, :no_mblocks}
 
-    {:ok, mb_hash} = :aec_headers.hash_header(:aec_blocks.to_micro_header(mblock))
+      list ->
+        {block_index, mblock, _mutations} =
+          list |> Enum.max_by(fn {block_index, _block, _mutations} -> block_index end)
 
-    {block_index, mb_hash, mblock}
+        {:ok, mb_hash} = :aec_headers.hash_header(:aec_blocks.to_micro_header(mblock))
+
+        {:ok, {block_index, mb_hash, mblock}}
+    end
   end
 
   defp exec_mem_mutations(empty_state, gens_mutations, from_height) do
```

Comment:
I would keep this file unchanged

Reply:
I can revert it to check if the issue with last_microblock persists after removing the new mutation